src/Microsoft.ML.OnnxConverter/OnnxContextImpl.cs

```diff
@@ -55,6 +56,28 @@ internal sealed class OnnxContextImpl : OnnxContext
             _modelVersion = modelVersion;
             _domain = domain;
             _onnxVersion = onnxVersion;
+            _optionalOpSetVersion = new OptionalOpSetVersion();
+        }
+
+        internal class OptionalOpSetVersion
+        {
```

Comment:
Since you are not using a default parameter, I don't think this mechanism is necessary.

Reply:
I use a default parameter in the first place and it works well locally, but it causes build errors in almost all PR pipelines.

Reply:
You are right that you shouldn't use a default parameter. I agree with the way you have implemented the API surface. There is one function which will export to the latest opset version and another function where the user can specify the opset version to export to. The second function doesn't need to have the opset version as a default parameter. And since you now don't have default parameters, you don't need the options class that was discussed.

From within OnnxExportExtensions.cs you can call `new OnnxContextImpl() with the latest opset version in one function and with the user specified opset version in the second function

---
In reply to: [432728618](https://github.com/dotnet/machinelearning/pull/5176#discussion_r432728618) [](ancestors = 432728618)